packages/block-editor/src/components/block-settings-menu/index.js

```diff
@@ -108,7 +108,10 @@ export function BlockSettingsMenu( { clientIds, onSelect } ) {
 							{ ! isLocked && (
 								<MenuItem
 									className="editor-block-settings-menu__control block-editor-block-settings-menu__control"
-									onClick={ onRemove }
+									onClick={ () => {
+										onClose();
```

Comment:
This does resolve the issue, but it makes the behaviour of this menu item inconsistent with the others.

Reply:
I think it's fine as the focus needs to be moved to the different block anyway.